### ISSUE TITLE
SelenideElement Condition should not throw element not found exception if the condition not match

### ISSUE THREAD

**Issue by Balilaci** [selenide/selenide#886]
## The problem
When I use the SelenideElement.ShouldHave(text/exactText("expectedText")) on a snackbar element it is bad if the process wait for the text if it is not matched, because than I get an element not found exception insted some kind of text not match exception.

## Details
If necessary, describe the problem you have been experiencing in more detail.

## Tell us about your environment
* **Selenide Version:4.10**
* **Chrome\Firefox\IE Version:**
* **Browser Driver Version:**
* **Selenium Version:**
* **OS Version:**

## Code To Reproduce Issue [ Good To Have ]
Please remember that with sample code it's easier to reproduce the bug.

**Comment by vinogradoff** [comment]
Could you please add snippet and logs explaining your problem?

I am not getting it otherwise

**Comment by Balilaci** [comment]
Here is a part of the HTML code that exists only for a few seconds.

`<div class="v-snack snackbar-style v-snack--active v-snack--bottom v-snack--right" id="sbr_notification">
	<div class="v-snack__wrapper success">
		<div class="v-snack__content">
			<div class="layout">
				<div class="flex snackbar-icon align-centervertically xs2 sm-1">
					<i aria-hidden="true" class="v-icon material-icons">
						edit
					</i>
				</div> 
				<div class="flex snackbar-text align-centervertically"> 
					Sikeres művelet!
				</div>
				<div class="spacer">
				</div> 
				<div class="flex align-centervertically xs2 sm-1">
					<button type="button" class="snackbar-btn v-btn v-btn--flat">
						<div class="v-btn__content">
							<i aria-hidden="true" class="v-icon material-icons">
								close
							</i>
						</div>
					</button>
				</div>
			</div>
		</div>
	</div>
</div>`

And here is a code that I had expected to work:

`$(byId("sbr_notification")).shouldHave(text(expectedMessage)).$("button").click();`

But I had to change it to this:

`		String snackbarText = $(byId("sbr_notification")).$(byClassName("snackbar-text")).getText();
		if (snackbarText != null && !snackbarText.equals(expectedMessage)) {
		    Screenshots.takeScreenShot(Selenide.class.getName(), Thread.currentThread().getName());
		    throw new DialogTextMismatch(snackbarText, expectedMessage);
		}
		try {
			$(byId("sbr_notification")).$("button").click();
		} catch (Exception e) {
			
		}
`
Lets say the snackbar has a different message, in this case I do not want to see that the element not found i n the logs, but I expect the element was found, but the text is different than it was expected.

**Comment by vinogradoff** [comment]
Separate it in two lines:

```
$("#sbr_notification").shouldHave(text(expectedMessage));
$("#sbr_notification").$("button").click();
```

should work just fine.

**Comment by asolntsev** [comment]
@Balilaci Does it solve your problem?

**Comment by asolntsev** [comment]
closing due to inactivity

**Comment by Balilaci** [comment]
Sorry, but I only get notification now.
So, I don't kept it in one line (I changed it a while), but there are planty of problems, and some of them are really rear and random. Probably because the SnackBar message not just pop-up, but it has a transition to appear and disappear.
So finally I use two do-while cycle.
The first checks the SnackBar and the SnackBarMessage already exists, and second (because some time even the SnackBarMessage already exists, but it's text returned nothing) I wait until the SnackBarMessage lenght greater than 0.
The thing is our developers use vuetify, which is developed by a team that probably do not give a thing about the test automation even today when it is so common.
They have so many strange solution that is difficult to automate.
Luckily I found out that the Selenide have a solution to register my own commands within it's own. It is a great feature that has a poor publication.
